op-devstack/dsl/ecotone_fees.go

```diff
@@ -58,24 +58,27 @@ func (ef *EcotoneFees) ValidateTransaction(from *EOA, to *EOA, amount *big.Int)
 	vaultsAfter := ef.getVaultBalances(client)
 	vaultIncreases := ef.calculateVaultIncreases(vaultsBefore, vaultsAfter)
 
-	l1Fee := big.NewInt(0)
-	if receipt.L1Fee != nil {
-		l1Fee = receipt.L1Fee
-	}
+	// In Ecotone, L1 fee includes both base fee and blob base fee components
+	l1Fee := vaultIncreases.L1FeeVault // Use actual vault increase as the source of truth
```

Comment:
Can we still check that the receipt included L1 fee is equal to the `vaultIncreases.L1FeeVault`?

Reply:
Hopefully addressed in [e0feddc](https://github.com/ethereum-optimism/optimism/pull/16817/commits/e0feddc48d33b660574f9e8bf6a96b4cf35c91b3). Please take a look and let me know if this addresses your feedback.

Reply:
Left comment at https://github.com/ethereum-optimism/optimism/pull/16817#discussion_r2297602747